### ISSUE TITLE
JSON hero does not allow you to open large JSON files

### ISSUE THREAD

**Issue by NobleDraconian** [jsonhero-io/jsonhero-web#42]
Currently, as of `4/29/22`, if you attempt to open a large JSON file (around 22MB), the site does not allow you to open it. It simply acts as if you selected no file!

Usecase : Using JSON hero to browse log files & other automatically generated datastructures.

**Comment by bolshoytoster** [comment]
@NobleDraconian are there  any errors in the console (`f12`)?

**Comment by ericallam** [comment]
Yup this is a known issue, and is related to how the Column View is implemented. We need to reimplement it using virtualization so we can scale the json documents. This is something that is definitely on the roadmap.

**Comment by ubuntudroid** [comment]
Sample JSON for you guy to try your implementation against: https://raw.githubusercontent.com/kareltucek/firmware/master/examples/mirroring_keymap.json

**Comment by matt-aitken** [comment]
@ericallam I've created [a branch with column item virtualization](https://github.com/jsonhero-io/jsonhero-web/tree/feature/column-view-virtualization). This is only part of the solution though.

Locally I'm getting `RangeError: Maximum call stack size exceeded` errors for very large files. I've found two separate causes for this:

1. Getting related paths, i.e. the `getRelatedPathsAtPath` function. Example: https://jsonplaceholder.typicode.com/photos 
2. Inferring the types. Example from @ubuntudroid above: https://raw.githubusercontent.com/kareltucek/firmware/master/examples/mirroring_keymap.json

**Comment by paulmwatson** [comment]
Great to hear this is on the roadmap as my first test of JSON hero was a 14mb GEOJSON file and it didn't work at all;
https://github.com/OpenUpSA/nu-land-parcel-map/blob/main/src/data/complete.json

**Comment by ericallam** [comment]
@paulmwatson thanks for sending us a nice large JSON file to test with!

**Comment by migdur** [comment]
Hello, is there any progress on this topic? I can also supply large json files (around 4.7MB) for testing/debugging purposes if needed. Thanks.

**Comment by jonaskuiler** [comment]
Hey @ericallam, is there anything I can do to help?

**Comment by arunmg007** [comment]
Facing the same issue @ericallam, is this in the roadmap to support large files?
